cmd/globalAlignment/globalAlignment.go

```diff
@@ -32,6 +55,13 @@ func main() {
 		log.Fatalf("error, unable to read .fa files, check for > symbol before each name and that each fasta entry is only one line. Check to make sure there are only four bases or N, globalAlignment is unable to use anything outside A-T-G-C-N.\n")
 	}
 
+	//raven added this block
+	numSeqOne := CountSeqIdx(flag.Arg(0))
```

Comment:
This is a good solution, but it might be even better if you didn't need to open the file twice.  It will work fine in most cases, but if someone pipes data into the program, you can not open the stream twice.
Another option you could think about is untangling the big line that calls EasyOpen and NextFasta together.  If you call easy open by itself and save the file handle as inputFileOne and inputFileTwo, you are then free to call fasta.NextFasta on that handle repeatedly.  I think you could then go two paths.  The quickest would be to call fasta.NextFasta a second time on that file handle.  Just like there is an error check the first time to make sure "done" is not true, the second time you would check to make sure that "done" is true.  The second option would be to have your CountSeqIdx function take the EasyReader (that in this case has already read the first sequence) instead of a filename as the input.  Then it could count how many "extra" sequences are in the file and give an error if this is not zero.

Reply:
I added the inputFile variables and changed the CountSeqIdx input to EasyReader. It worked, but I will try Dan's len(faOne) suggestion since that might be more efficient